packages/better-auth/src/plugins/jwt/sign.ts

```diff
@@ -1,42 +1,100 @@
-import { importJWK, SignJWT } from "jose";
+import {
+	importJWK,
+	exportJWK,
+	generateKeyPair,
+	SignJWT,
+	type JWTPayload,
+} from "jose";
 import type { GenericEndpointContext } from "../../types";
 import { BetterAuthError } from "../../error";
 import { symmetricDecrypt, symmetricEncrypt } from "../../crypto";
-import { generateExportedKeyPair, type JwtOptions } from ".";
+import { getJwtPlugin, toExpJWT } from "./utils";
 import type { Jwk } from "./schema";
 import { getJwksAdapter } from "./adapter";
+import type { JwtPluginOptions } from "./types";
 
-export async function getJwtToken(
+/**
+ * Signs a payload in jwt format.
+ * For security, exp will be generated (from iat or current time) if not supplied.
+ *
+ * @internal - SCOPED TO PLUGIN. Use signJwt for usage in other plugins.
+ *
+ * @param ctx - endpoint context
+ * @param payload - payload to sign
+ */
+export async function signJwtPayload(
 	ctx: GenericEndpointContext,
-	options?: JwtOptions,
+	payload: JWTPayload,
+	options?: JwtPluginOptions,
 ) {
+	// Iat strict checks
+	const nowSeconds = Math.floor(Date.now() / 1000);
+	const iat = payload.iat;
+	if (iat && iat > nowSeconds) {
+		throw new BetterAuthError("unable to set a future iat time");
+	}
+
+	// Exp safety checks
+	let exp = payload.exp;
+	const allowLargerExpTime = options?.jwt?.allowLongerExpTime;
+	const defaultExp = toExpJWT(options?.jwt?.expirationTime ?? "1h", iat ?? nowSeconds);
+	if (!allowLargerExpTime && exp && exp > defaultExp) {
+		throw new BetterAuthError("unable to set future exp time");
+	}
+	exp = exp ?? defaultExp;
+
+	// Nbf strict checks
+	const nbf = payload.nbf;
+	if (nbf &&  !((iat && nbf < iat) || nbf > exp)) {
```

Comment:
Implementation Note: Using `iat && nbf < iat` instead of `nbf < (iat ?? nowSeconds)` because nbf can only be checked against iat if sent, nowSeconds is local and may fail if used in this comparison.